test/test_puma_server.rb

```diff
@@ -786,11 +786,23 @@ def test_chunked_keep_alive_two_back_to_back
     @server.run
 
     sock = TCPSocket.new @host, @server.connected_port
-    sock << "GET / HTTP/1.1\r\nConnection: Keep-Alive\r\nTransfer-Encoding: chunked\r\n\r\n1\r\nh\r\n4\r\nello\r\n0\r\n\r\n"
+    sock << "GET / HTTP/1.1\r\nConnection: Keep-Alive\r\nTransfer-Encoding: chunked\r\n\r\n1\r\nh\r\n4\r\nello\r\n0\r\n"
+
+    last_crlf_written = false
+    last_crlf_writer = Thread.new do
+      sleep 0.1
+      sock << "\r"
+      sleep 0.1
+      sock << "\n"
+      last_crlf_written = true
+    end
 
     h = header(sock)
     assert_equal ["HTTP/1.1 200 OK", "Content-Length: 0"], h
     assert_equal "hello", body
+    assert_equal true, last_crlf_written
+
+    last_crlf_writer.join
```

Comment:
Doesn't this line have to be *before* `assert last_crlf_written`?

Reply:
We need to put this here.
This tests that Puma must wait the last CRLF to complete HTTP body processing.
If we put this before `assert_equal true, last_crlf_written`, the assertion is passed even if Puma doesn't wait the last CRLF. (This is the current behavior. And this causes this problem.)